### ISSUE TITLE
Spaces and Parenthesis removed from Extension Name in Error Message

### ISSUE THREAD

**Issue by imanickam** [joomla/joomla-cms#35946]
### Steps to reproduce the issue

(a) Log into administrator back-end
(b) Install the French language
(c) Go to screen "Extensions: Manage" (System => Manage -> Extensions)
(d) Select the extension that has the Name as "French (fr-FR)", Location as "Site", and Type as "Language"
(e) Press the button "Uninstall"

Note:
French language has been used for illustration only.  This behavior is exhibited in all the language packs.

### Expected result

The error message should contain the full name of the extension 
The French (fr-FR) extension is part of a package which does not allow individual extensions to be uninstalled.

### Actual result

The Frenchfr-FR extension is part of a package which does not allow individual extensions to be uninstalled.

![screen shot 2021-10-30 at 08 41 04](https://issues.joomla.org/uploads/1/76f87cddc5780ed251cbf58ad3649ba2.jpg)

Notice that the spaces and parenthesis are removed from the extension name.  You can notice this behavior in the following example as well.

![screen shot 2021-10-30 at 08 41 49](https://issues.joomla.org/uploads/1/55184e2192d5dbdc99a1424a0fc02b00.jpg)

### System information (as much as possible)

Joomla! 4.0.4

### Additional comments

This may not be applicable only to language packs.  This behavior would be exhibited in any extension that is a part of a package where the tag 
```
<blockChildUninstall>true</blockChildUninstall>
```
is available in the package's xml file.

**Comment by infograf768** [comment]
Confirmed.

**Comment by infograf768** [comment]
In fact, the extension language name is saved a such in the db.
<img width="807" alt="Screen Shot 2021-10-31 at 10 04 36" src="https://user-images.githubusercontent.com/869724/139575562-96fc8ba1-f61a-47a2-ad68-989ced27a25c.png">


<img width="743" alt="Screen Shot 2021-10-31 at 10 07 00" src="https://user-images.githubusercontent.com/869724/139575633-5109b7f8-3c2a-42cd-a1d5-c843f9fafa42.png">


<img width="291" alt="Screen Shot 2021-10-31 at 10 07 40" src="https://user-images.githubusercontent.com/869724/139575642-6fb5fead-055c-4d6f-b4b1-57017d8fd1d5.png">

EDIT/NOTE: same behavior in 3.x

**Comment by infograf768** [comment]
This comes from this code in the language adapter

https://github.com/joomla/joomla-cms/blob/65d9dc74dad2630d14cd6c8041e0dd4f4f92147d/libraries/src/Installer/Adapter/LanguageAdapter.php#L307

`->clean` the manifest name takes off the parenthesis and spaces.

**Comment by richard67** [comment]
No idea why the "cmd" filter is used here. Maybe we should use "string"? Or "html"?

**Comment by HLeithner** [comment]
IIRC the name is parsed by JText that's the reason why spaces are not allowed, normally a extension name is com_xxx and in the component ini you have a language string com_xxx="The XXX component" I would expect the same is true for language packages.

**Comment by imanickam** [comment]
Thanks for looking into this.

FYI - The same behavior is exhibited in 3.10 as well.

**Comment by infograf768** [comment]
> IIRC the name is parsed by JText that's the reason why spaces are not allowed, normally a extension name is com_xxx and in the component ini you have a language string com_xxx="The XXX component" I would expect the same is true for language packages.

Languages names (for clients) are an exception. Indeed they pass through `Text::_()` as all extensions  when displayed in the Extensions Manager, but **they should never be translated but always presented in English**.
The best demonstration is English (en-GB) which in the db is presented in full. Package and clients.

The other languages **package** names are also saved in db in full. No `clean` there but only the manifest name. Only the client languages are not.

<img width="416" alt="Screen Shot 2021-11-01 at 05 51 19" src="https://user-images.githubusercontent.com/869724/139623317-ea460991-0a3b-4f36-966a-fe726beb4f32.png">

In the Extensions Manager, 
![Screen Shot 2021-11-01 at 05 52 36](https://user-images.githubusercontent.com/869724/139623359-c5f5a023-0022-48df-b444-ef99c79913f6.png)

`Text::_()` is not used in the Installed languages view and it is fine. (Italiano is a wrong name. Should be Italian...)

![Screen Shot 2021-11-01 at 06 03 38](https://user-images.githubusercontent.com/869724/139624059-c723f3a9-6c74-4af0-a170-d135d86b5afe.png)

Therefore,  there is no reason to install client languages with `cmd`.

I tested changing `cmd` to `string` and it now works fine.
`$this->name = InputFilter::getInstance()->clean((string) $this->getManifest()->name, 'string');`

<img width="963" alt="Screen Shot 2021-11-01 at 06 11 53" src="https://user-images.githubusercontent.com/869724/139624684-c60cfc8c-24e2-4246-ae57-917883fd6416.png">

**Comment by richard67** [comment]
Hmm, if we change the filter for the name and the names are stored in database after filtering, then after changing the filter to "string" the name will not match anymore to what is stored in database. So it would need to fix the values in database with values from the manifests for all extensions of whichever type when updating from a version without the change to a version with the change from a potential fix.

**Comment by infograf768** [comment]
I do not think so.
1. It concerns only the LanguageAdapter.php i.e. languages extensions
2. We would also have to change 'cmd' to 'string' in the update method. 
3. I tested.

I installed the French pack https://downloads.joomla.org/language-packs/translations-joomla4/downloads/joomla4-french/4-0-3-1 

then I modified the LanguageAdapter to use 'string' and went to install languages: it proposed pack 4.0.4.1.
All went fine.

There is a good reason for that, the adapter does not check for equal names but only tag, type and client
```
// Update an entry to the extension table
		$row = Table::getInstance('extension');
		$eid = $row->find(array('element' => $this->tag, 'type' => 'language', 'client_id' => $clientId));

		if ($eid)
		{
			$row->load($eid);
		}
		else
		{
			// Set the defaults

			// There is no folder for language
			$row->set('folder', '');
			$row->set('enabled', 1);
			$row->set('protected', 0);
			$row->set('access', 0);
			$row->set('client_id', $clientId);
			$row->set('params', $this->parent->getParams());
		}

		$row->set('name', $this->name);
```	
$this->name is taken from a bit above in the code. It is the new pack manifest:

```
// Get the language name
		// Set the extensions name
		$name = (string) $this->getManifest()->name;
		$name = InputFilter::getInstance()->clean($name, 'string'); // which I just changed to `string`
		$this->name = $name;
```	
As for English it is already fine as English (en-GB) in the db.

**Comment by richard67** [comment]
Pull request for 3.10 is #35980 . Please test. Later when it has been merged, it will go into 4.0-dev with the next regular upmerge after that.

But maybe I make another PR with the same change for 4.0-dev for human testers.

**Comment by richard67** [comment]
Pull request for Joomla 4 is #35981 .

@imanickam Please test. If you can test with both 3.10.x and 4.0.y, test #35980 , too . Thanks in advance.

Closing as having a pull request for each 3.10 and 4.0.
